mobile/lib/modules/asset_viewer/views/gallery_viewer.dart

```diff
@@ -332,6 +351,26 @@ class GalleryViewerPage extends HookConsumerWidget {
       );
     }
 
+    handleDownload() {
+      if (asset().isLocal) {
+        return;
+      }
+      if (asset().isOffline) {
+        ImmichToast.show(
+          durationInSecond: 1,
+          context: context,
+          msg: 'asset_action_share_err_offline'.tr(),
+          gravity: ToastGravity.BOTTOM,
+        );
+        return;
+      }
+
+      ref.watch(imageViewerStateProvider.notifier).downloadAsset(
```

Comment:
Sorry for the spam: use read instead of watch in callbacks?

Reply:
Ah, not a problem 😄 Will update all such instances soon. Thank you!